File: sdk/javascript/paissive_income_sdk/services/agent-team.js
/**
 * Agent Team service for the pAIssive Income API.
 * 
 * This module provides a service for interacting with the agent team endpoints.
 */

const BaseService = require('./base');

/**
 * Agent Team service.
 * @extends BaseService
 */
class AgentTeamService extends BaseService {
  /**
   * Create an agent team.
   * 
   * @param {Object} data - Team creation data
   * @param {string} data.name - Team name
   * @param {string} data.description - Team description
   * @param {Array} data.agents - List of agent IDs or configurations
   * @returns {Promise<Object>} Created team
   */
  async createTeam(data) {
    return this._post('agent-team/teams', data);
  }
  
  /**
   * Get all agent teams.
   * 
   * @returns {Promise<Object>} List of agent teams
   */
  async getTeams() {
    return this._get('agent-team/teams');
  }
  
  /**
   * Get details for a specific agent team.
   * 
   * @param {string} teamId - Team ID
   * @returns {Promise<Object>} Team details
   */
  async getTeam(teamId) {
    return this._get(`agent-team/teams/${teamId}`);
  }
  
  /**
   * Update an agent team.
   * 
   * @param {string} teamId - Team ID
   * @param {Object} data - Updated team data
   * @returns {Promise<Object>} Updated team
   */
  async updateTeam(teamId, data) {
    return this._put(`agent-team/teams/${teamId}`, data);
  }
  
  /**
   * Delete an agent team.
   * 
   * @param {string} teamId - Team ID
   * @returns {Promise<Object>} Result of the deletion
   */
  async deleteTeam(teamId) {
    return this._delete(`agent-team/teams/${teamId}`);
  }
  
  /**
   * Get all available agents.
   * 
   * @returns {Promise<Object>} List of agents
   */
  async getAgents() {
    return this._get('agent-team/agents');
  }
  
  /**
   * Get details for a specific agent.
   * 
   * @param {string} agentId - Agent ID
   * @returns {Promise<Object>} Agent details
   */
  async getAgent(agentId) {
    return this._get(`agent-team/agents/${agentId}`);
  }
  
  /**
   * Get all available workflows.
   * 
   * @returns {Promise<Object>} List of workflows
   */
  async getWorkflows() {
    return this._get('agent-team/workflows');
  }
  
  /**
   * Run a workflow with an agent team.
   * 
   * @param {Object} data - Workflow execution data
   * @param {string} data.team_id - Team ID
   * @param {string} data.workflow_id - Workflow ID
   * @param {Object} data.inputs - Workflow inputs
   * @returns {Promise<Object>} Workflow execution results
   */
  async runWorkflow(data) {
    return this._post('agent-team/workflows/run', data);
  }
}

module.exports = AgentTeamService;

File: sdk/python/paissive_income_sdk/services/agent_team.py
"""
Agent Team service for the pAIssive Income API.

This module provides a service for interacting with the agent team endpoints.
"""

from typing import Any, Dict

from .base import BaseService

class AgentTeamService(BaseService):
    """
    Agent Team service.
    """

    def create_team(self, data: Dict[str, Any]) -> Dict[str, Any]:
        """
        Create an agent team.

        Args:
            data: Team creation data
                - name: Team name
                - description: Team description
                - agents: List of agent IDs or configurations

        Returns:
            Created team
        """
        return self._post("agent - team / teams", data)

    def get_teams(self) -> Dict[str, Any]:
        """
        Get all agent teams.

        Returns:
            List of agent teams
        """
        return self._get("agent - team / teams")

    def get_team(self, team_id: str) -> Dict[str, Any]:
        """
        Get details for a specific agent team.

        Args:
            team_id: Team ID

        Returns:
            Team details
        """
        return self._get(f"agent - team / teams/{team_id}")

    def update_team(self, team_id: str, data: Dict[str, Any]) -> Dict[str, Any]:
        """
        Update an agent team.

        Args:
            team_id: Team ID
            data: Updated team data

        Returns:
            Updated team
        """
        return self._put(f"agent - team / teams/{team_id}", data)

    def delete_team(self, team_id: str) -> Dict[str, Any]:
        """
        Delete an agent team.

        Args:
            team_id: Team ID

        Returns:
            Result of the deletion
        """
        return self._delete(f"agent - team / teams/{team_id}")

    def get_agents(self) -> Dict[str, Any]:
        """
        Get all available agents.

        Returns:
            List of agents
        """
        return self._get("agent - team / agents")

    def get_agent(self, agent_id: str) -> Dict[str, Any]:
        """
        Get details for a specific agent.

        Args:
            agent_id: Agent ID

        Returns:
            Agent details
        """
        return self._get(f"agent - team / agents/{agent_id}")

    def get_workflows(self) -> Dict[str, Any]:
        """
        Get all available workflows.

        Returns:
            List of workflows
        """
        return self._get("agent - team / workflows")

    def run_workflow(self, data: Dict[str, Any]) -> Dict[str, Any]:
        """
        Run a workflow with an agent team.

        Args:
            data: Workflow execution data
                - team_id: Team ID
                - workflow_id: Workflow ID
                - inputs: Workflow inputs

        Returns:
            Workflow execution results
        """
        return self._post("agent - team / workflows / run", data)
